File: html/app/models/lead.py
from sqlalchemy import Column, Integer, String, DateTime, ForeignKey, Enum, Boolean, func
from sqlalchemy.orm import relationship
from datetime import datetime
import enum
from .base import Base

class LeadStatus(str, enum.Enum):
    NEW = "new"
    CALL = "call"
    TRIAL = "trial"
    OFFER = "offer"
    DEAL = "deal"
    SUCCESS = "success"
    REJECT = "reject"

class Lead(Base):
    __tablename__ = "leads"

    id = Column(Integer, primary_key=True, index=True)
    name = Column(String, index=True)
    phone = Column(String, index=True)
    age = Column(Integer, nullable=True)  # New field
    next_contact_date = Column(DateTime, nullable=True)  # New field
    status = Column(String, default=LeadStatus.NEW)
    source = Column(String, nullable=True)
    notes = Column(String, nullable=True)
    rejection_reason = Column(String, nullable=True)
    first_call_at = Column(DateTime, nullable=True)
    first_trial_at = Column(DateTime, nullable=True)
    
    created_at = Column(DateTime, default=datetime.utcnow)
    updated_at = Column(DateTime, default=datetime.utcnow, onupdate=datetime.utcnow)
    
    created_by_id = Column(Integer, ForeignKey("users.id"), nullable=True)
    created_by = relationship("User", foreign_keys=[created_by_id])
    
    responsible_id = Column(Integer, ForeignKey("users.id"), nullable=True)
    responsible = relationship("User", foreign_keys=[responsible_id])


class LeadTask(Base):
    __tablename__ = "lead_tasks"

    id = Column(Integer, primary_key=True, index=True)
    lead_id = Column(Integer, ForeignKey("leads.id", ondelete="CASCADE"), nullable=False, index=True)
    title = Column(String, nullable=False)
    due_date = Column(DateTime, nullable=True)
    completed = Column(Boolean, default=False, nullable=False)
    created_at = Column(DateTime, server_default=func.now(), nullable=False)

    lead = relationship("Lead", backref="tasks")

File: html/football_academy_app/lib/models/lead.dart
class Lead {
  final int id;
  final String name;
  final String phone;
  final int? age;
  final DateTime? nextContactDate;
  final String status;
  final String? source;
  final String? notes;
  final int? responsibleId;
  final DateTime? createdAt;
  final DateTime? updatedAt;

  Lead({
    required this.id,
    required this.name,
    required this.phone,
    this.age,
    this.nextContactDate,
    required this.status,
    this.source,
    this.notes,
    this.responsibleId,
    this.createdAt,
    this.updatedAt,
  });

  factory Lead.fromJson(Map<String, dynamic> json) {
    return Lead(
      id: json['id'],
      name: json['name'],
      phone: json['phone'],
      age: json['age'],
      nextContactDate: json['next_contact_date'] != null
          ? DateTime.parse(json['next_contact_date'])
          : null,
      status: json['status'] ?? 'new',
      source: json['source'],
      notes: json['notes'],
      responsibleId: json['responsible_id'],
      createdAt: json['created_at'] != null
          ? DateTime.parse(json['created_at'])
          : null,
      updatedAt: json['updated_at'] != null
          ? DateTime.parse(json['updated_at'])
          : null,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'phone': phone,
      'age': age,
      'next_contact_date': nextContactDate?.toIso8601String(),
      'status': status,
      'source': source,
      'notes': notes,
      'responsible_id': responsibleId,
      'created_at': createdAt?.toIso8601String(),
      'updated_at': updatedAt?.toIso8601String(),
    };
  }
}
